examples/audio/audio_looping.rs

```diff
@@ -0,0 +1,46 @@
+use bevy::prelude::*;
+
+/// This example illustrates how to load and play an audio file on repeat.
+
+struct OverworldThemeTimer(Timer);
+
+struct OverworldTheme {
+    track: Handle<AudioSource>,
+}
+
+const THEME_LENGTH: f32 = 13.;
+
+fn main() {
+    App::new()
+        .add_plugins(DefaultPlugins)
+        .insert_resource(OverworldThemeTimer(Timer::from_seconds(THEME_LENGTH, true)))
+        .init_resource::<OverworldTheme>()
+        .add_startup_system(play_theme.system())
+        .add_system(repeat_theme.system())
+        .run();
+}
+
+impl FromWorld for OverworldTheme {
+    fn from_world(world: &mut World) -> Self {
+        let world = world.cell();
+        let asset_server = world.get_resource_mut::<AssetServer>().unwrap();
+        OverworldTheme {
+            track: asset_server.load("sounds/loop_me_melancholy.mp3"),
+        }
+    }
+}
+
+fn play_theme(audio: Res<Audio>, theme: Res<OverworldTheme>) {
+    audio.play(theme.track.clone());
```

Comment:
I think this will be clearer if it's just inlined here, perhaps with a comment.

Reply:
Oh cool! You're reusing this function as both a function and a system! Never mind!